packages/core/src/_common/base/api_middleware.js

```diff
@@ -1,9 +1,94 @@
+const datadogLogs = require('@datadog/browser-logs').datadogLogs;
+const formatDate = require('@deriv/shared').formatDate;
+const formatTime = require('@deriv/shared').formatTime;
+
+const datadog_client_token_logs = process.env.DATADOG_CLIENT_TOKEN_LOGS ?? '';
+const is_production = process.env.CIRCLE_JOB === 'release_production';
+const is_staging = process.env.CIRCLE_JOB === 'release_staging';
+const datadog_session_sample_rate = +process.env.DATADOG_SESSION_SAMPLE_RATE_LOGS ?? 1;
+let datadog_version = '';
+let datadog_env = '';
+const REQUESTS = [
```

Comment:
if there is any requests which p2p makes continuously in background then, then we need to add some setTimeout condition to restrict the logs send to datadog, or else this looks fine